### DISCUSSION TITLE
Standalone hangs, maya version works, why ?

### DISCUSSION THREAD

**Discussion by maurizio1974** [nxt-dev/nxt_editor#136] [General]
First of all guys I can;t thank you enough for this tool, it is amazing and I am sure it can only get more powerfull.
I did follow the installation videos and in a virtual environment I got nxt-editor standalone
installed don fedora 31 with python 3.7.9, but when I open it it hangs the machine for 2 or 3 seconds then after almost a minute it opens a huge window with nothing inside and stays there. Oddly if I run the maya plugin the editor opens fine and it doesn't cause any issue.

Any clue ?

Thank you in advance

**Comment by ImLucasBrown** [comment]
Can you post the output of `pip list` in that virtual env?
We've had reports that some versions of PySide2 have issues building the qresources.

**Reply by maurizio1974** [reply]
here the list printed of that virtual env

Package    Version
---------- -------
nxt-core   0.9.1
nxt-editor 3.5.1
pip        20.3.3
PySide2    5.11.1
Qt.py      1.1.0
setuptools 41.6.0

**Reply by ImLucasBrown** [reply]
Hmm odd, I booted up a Fedora 33 VM with the same env you listed and wasn't able to repro the issue.
Does anything print out in the terminal when you launch with `nxt ui` (aside from the git error)? I noticed a Qt error on the first launch that seemed to resolve itself on the second launch.

**Reply by maurizio1974** [reply]
let me check, but I couldn't see anything weird. COuld it be something related to the nvidia card, cause on another machine same setup but with integrated GPU is working fine

**Reply by maurizio1974** [reply]
nope, the only thing I am getting is the git error.
I also tried on a VM of fedora31, and centos 8, and there it doesn't even find a suitable version to install for some reasons with pip. Or at least that is what it says when I launch the pip command to install in the virtual environment

**Comment by MichaelAldrich** [comment]
I booted fedora31 from a live disk this morning, updated python to 3.7.9, and ran an install that worked as expected. Here is the log from that: https://pastebin.com/jt6Jb38y
We haven't seen anything problems stemming from nvidia cards. This test was done on a gtx980, and we've had mixed gpu usage for the last year without any indicator of it as a problem.

I'm glad it's working for you in maya @maurizio1974 , but I'm not sure what else we can do to test or help your standalone from our side. When it hangs and opens a blank window, is there any console output you can share?

**Reply by maurizio1974** [reply]
Hi Michael, I will keep digging and see why I am getting this issue, no output on the shell beside the git error, unfortunately. I will keep you posted in case I find out something more but looks definitely that it is on my side
Thank you

**Reply by maurizio1974** [reply]
> I booted fedora31 from a live disk this morning, updated python to 3.7.9, and ran an install that worked as expected. Here is the log from that: https://pastebin.com/jt6Jb38y
> We haven't seen anything problems stemming from nvidia cards. This test was done on a gtx980, and we've had mixed gpu usage for the last year without any indicator of it as a problem.
> 
> I'm glad it's working for you in maya @maurizio1974 , but I'm not sure what else we can do to test or help your standalone from our side. When it hangs and opens a blank window, is there any console output you can share?

Ok So I did the install again from your paste-bin and I got the same result, but  I noticed that I can run the command 'nxt test' and the result is attached. hopefully, this helps

[init_test.zip](https://github.com/nxt-dev/nxt_editor/files/5838201/init_test.zip)

**Reply by MichaelAldrich** [reply]
Thanks for the reference, do I understand right that it's working for you now?

The failing test looks like it's due to test machinery, not a failure to install.

**Reply by maurizio1974** [reply]
nope still not working, I thought the init test would have helped. Maybe it is time for a clean reinstall of my system.
I will keep digging

cheers

**Reply by maurizio1974** [reply]
Hi Michael

so I am able to reproduce the issue, but not specific log information.
I installed a clean version of fedora and then nxt_editor which was working just fine, as soon as I install
the Nvidia driver ( Tried on two different machines ) then the hangs happen again. I noticed that
on my laptop set up with Optimus it works fine, I guess it picks the normal igpu maybe?
On the desktop with only discrete graphics it doesn;t work with the Nvidia drivers, but if I remove them, then it works just fine
I also tried on a virtual machine and there it works fine, but unless you set up a GPU pass-through, which I didn't do then the Nvidia card is not used so we go back to the possible igpu situation.
Is there a way to maybe launch the ui with no OpenGL?

**Reply by MichaelAldrich** [reply]
Thanks for doing such thorough research into reproducing it. 

We don't have any tools for launching with no OpenGL, however having this repro is very helpful. I'd like to try to repro these steps.
>  a clean version of fedora and then nxt_editor which was working just fine, as soon as I install
the Nvidia driver ( Tried on two different machines ) then the hangs happen again.

Can you tell me what you're using to install the drivers and the version numbers you're seeing it with?
Once I can reproduce the problem I can find a solution.

**Reply by maurizio1974** [reply]
Hi Michael

To install the drivers I tried two different way:
- The manual way with drivers from their web site ( 460.39 )
- the akmod-nvidia from the rpm-fusion repository that does the above steps automatically
there are both guides explained in this article
https://www.fosslinux.com/44725/how-to-install-nvidia-drivers-on-fedora-workstation.htm

I tired on an old card 960 gtx and a new 2080 super, both on fedora-mate 31

Thank you

**Reply by maurizio1974** [reply]
so I finnaly figured out why was not working, it has to do with the resolution of the monitor.
I have a 49 inch samsung monitor at 52120 x 1440 and at that res it hangs, but if I change the resolution
at 3840 x 1080 then it works fine ( just very slow waiting for the rpc server  any solution on that ?)
Funny enough this happens only on the machine with discrete nvidia graphic cards and also after I opened it
if I go back to the higher resolution everything works fine.
Not sure if that gives you a clue on what can be done if anything can be done, just passing along, for the time being 
I am jsut switching the res at launch time

cheers

**Reply by ImLucasBrown** [reply]
Glad to hear the misty is somewhat solved, I will try reproing with a huge resolution like that, its possible its something to do with Qt's high dpi setup.
As far as the RPC server goes you can launch with the `-no-rpc` flag if you don't need multi context functionality.

**Reply by maurizio1974** [reply]
yes I think so to that is it is something to do with Qt.
no matter what I am all good to go 

thx
